cmd/agentctl/commands/dump.go

```diff
@@ -16,50 +16,41 @@ package commands
 
 import (
 	"bytes"
-	"encoding/json"
+	"context"
 	"fmt"
-	"net/url"
+	"io"
 	"os"
-	"reflect"
 	"sort"
 	"strings"
 	"text/tabwriter"
 
-	"github.com/gogo/protobuf/jsonpb"
 	"github.com/gogo/protobuf/proto"
 	"github.com/spf13/cobra"
 
+	"github.com/ligato/vpp-agent/api/types"
+	agentcli "github.com/ligato/vpp-agent/cmd/agentctl/cli"
 	"github.com/ligato/vpp-agent/plugins/kvscheduler/api"
 )
 
-func NewDumpCommand(cli *AgentCli) *cobra.Command {
+func NewDumpCommand(cli agentcli.Cli) *cobra.Command {
 	var opts DumpOptions
 
 	cmd := &cobra.Command{
-		Use:     "dump MODEL",
-		Aliases: []string{"d"},
-		Short:   "Dump running state",
+		Use:   "dump MODEL",
+		Short: "Dump running state",
 		Example: `
  To dump VPP interfaces run:
-  $ agentctl dump vpp.interfaces
+  $ {{.CommandPath}} vpp.interfaces
 
  To use different dump view use --view flag:
-  $ agentctl dump --view=NB vpp.interfaces
-
- For a list of all supported models that can be dumped run:
-  $ agentctl model list
-
- To specify the HTTP address of the agent use --host flag:
-  $ agentctl --host 172.17.0.3 dump vpp.interfaces
-`,
+  $ {{.CommandPath}} --view=NB vpp.interfaces`,
 		Args: cobra.ExactArgs(1),
-		Run: func(cmd *cobra.Command, args []string) {
+		RunE: func(cmd *cobra.Command, args []string) error {
 			opts.Models = args
```

Comment:
args will never have more than one element, because of `Args: cobra.ExactArgs(1),`. I think it would be better to assign like `opts.Model = args[0]` and not use additional variable in `runDump()`